### PR TITLE
ENH: Add innovations algorithm

### PR DESCRIPTION
**Pull request by bashtage**
Add innovations algorithm to convert acov to MA
Add filter to filter innovations usign output of ia

### CODE REVIEW

**Comment by josef-pkt** [comment]
looks good based on a quick skimming
closes #2593

**Comment by josef-pkt** [comment]
sorry, pushed the wrong button (not enough coffee yet

File: statsmodels/compat/pandas.py [hunk 1]
```diff
@@ -29,3 +29,12 @@ def sort_values(df, *args, **kwargs):
 
     data_klasses = (pandas.Series, pandas.DataFrame, pandas.Panel,
                     pandas.WidePanel)
+
+try:
+    import pandas.testing as testing
```
Comment (jbrockmendel) [inline statsmodels/compat/pandas.py]: What is the version number cutoff at which this becomes unnecessary?

**Comment by bashtage** [comment]
0.20

File: statsmodels/tsa/tests/test_stattools.py [hunk 2]
```diff
@@ -714,3 +714,78 @@ def test_pacf_burg_error():
...
+    ma = -0.9
+    acovf = np.array([1 + ma ** 2, ma])
+    theta, _ = innovations_algo(acovf, nobs=4)
+    e = np.random.randn(5)
+    endog = e[1:] + ma * e[:-1]
+    resid = innovations_filter(endog, theta)
+    expected = [endog[0]]
+    for i in range(1, 4):
+        expected.append(endog[i] + theta[i, 0] * expected[-1])
+    print(expected)
```
Comment (josef-pkt) [inline statsmodels/tsa/tests/test_stattools.py]: there is still a print left

**Comment by josef-pkt** [comment]
separated into two commits looks good/useful
I only superficially skimmed the algorithm, it might be a bit slow for large nobs and n=nobs

LGTM, AFAICS

**Comment by bashtage** [comment]
print is finally gone.

**Comment by bashtage** [comment]
It is not that bad for fairly long time series - I tested numba and it could be trivially sped up by a factor of 250-400 with a simple @numba.jit.

**Comment by josef-pkt** [comment]
I think we should add numba as optional requirement with several cases to gain experience with it in 0.11, it's a bit late for 0.10.

**Comment by josef-pkt** [comment]
merge when green (ETA 12 to 13 hours, i.e. not today)

**Comment by bashtage** [comment]
@josef-pkt green.

**Comment by josef-pkt** [comment]
@bashtage Thank you

**Pull request merged**